src/shared/utils/exportData/getSubscales.ts

```diff
@@ -1,11 +1,140 @@
+  DecryptedSexAnswer,
+  DecryptedSingleSelectionAnswer,
+  DecryptedSliderAnswer,
+} from 'modules/Dashboard/features/RespondentData/RespondentDataReview/RespondentDataReview.types';
+import { ElementType } from 'modules/Builder/features/SaveAndPublish/SaveAndPublish.types';
 
-export const getSubscales = (subscales?: ActivitySettingsSubscale[]) =>
-  subscales?.reduce((acc: { [key: string]: string | Record<string, string>[] }, item) => {
-    acc[item.name] = '1';
-    if (item?.subscaleTableData) {
-      acc[`Optional text for ${item.name}`] = item.subscaleTableData;
+import { getObjectFromList } from '../builderHelpers';
+import { createArrayFromMinToMax } from '../array';
+
+export const getSubScaleScore = (subscalesSum: number, type: SubscaleTotalScore, length: number) =>
+  type === SubscaleTotalScore.Sum ? subscalesSum : subscalesSum / length;
+
+export const parseSex = (sex: string) => (sex === 'M' ? '1' : '2');
+
+export const calcScores = <T>(
+  data: ActivitySettingsSubscale,
+  activityItems: Record<string, T & { answer: AnswerDTO; activityItem: Item }>,
+  subscalesObject: Record<string, ActivitySettingsSubscale>,
+): { score: number; optionText: string } => {
+  const sumScore = data.items.reduce((acc, item) => {
+    if (!item?.type) {
+      return acc;
+    }
+
+    if (item.type === ElementType.Subscale) {
+      return acc + calcScores(subscalesObject[item.name], activityItems, subscalesObject).score;
+    }
+
+    const answer = activityItems[item.name].answer as
+      | DecryptedMultiSelectionAnswer
+      | DecryptedSingleSelectionAnswer
+      | DecryptedSliderAnswer;
+    const typedOptions = activityItems[item.name].activityItem
+      .responseValues as SingleAndMultipleSelectItemResponseValues & SliderItemResponseValues;
+    let value = 0;
+
+    if (typedOptions?.options?.length) {
+      const scoresObject = typedOptions.options?.reduce((acc: ScoresObject, item) => {
+        if (item?.value && item?.score) {
+          acc[item.value as keyof ScoresObject] = item.score;
+        }
+
+        return acc;
+      }, {});
+
+      if (Array.isArray(answer?.value)) {
+        value = answer.value?.reduce((res: number, item) => res + scoresObject[item], 0);
+      } else {
+        value = scoresObject[answer?.value] || 0;
+      }
+    }
+
+    if (typedOptions?.minValue && typedOptions?.scores?.length) {
+      const min = typedOptions.minValue;
+      const max = typedOptions.maxValue;
+      const scores = typedOptions.scores;
+      const options = createArrayFromMinToMax(min, max);
+
+      value = scores[options.findIndex((item) => item === answer.value)] || 0;
+    }
+
+    return acc + value;
+  }, 0);
+
+  const calculetedScore = getSubScaleScore(sumScore, data.scoring, data.items.length);
+
+  if (data?.subscaleTableData) {
+    const subscaleTableDataItem = data.subscaleTableData?.find(({ sex, age, rawScore }) => {
+      const genderAnswer = activityItems.gender_screen?.answer as DecryptedSexAnswer;
```

Comment:
please add enum for 'gender_screen' and `age_screen` and use it for `name` properties in genderItem and ageItem in src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.const.ts